src/components/input/input.vue

```diff
@@ -0,0 +1,208 @@
+<script setup lang="ts">
+import {
+  defineEmits,
+  defineAsyncComponent,
+  ref,
+  computed,
+  watch,
+  useSlots,
+  type Ref,
+} from 'vue';
+import {
+  inputSize,
+  defaultVariantMixin,
+  inputVariant,
+} from '@/helpers/mixins/jsMixins';
+import('@/components/button/button.vue');
+
+type InputProps = {
+  text: string
+  variant?: string
+  validationText?: string
+  isValid?: boolean | null
+  placeholder?: string | undefined
+  disabled?: boolean
+  modelValue?: string | null
+  helpertext?: string
+  size?: string
+};
+
+const slots = useSlots();
+
+const emit = defineEmits(['change', 'update:modelValue']);
+
+const props = withDefaults(defineProps<InputProps>(), {
+  validationText: '',
+  isValid: null,
+  placeholder: undefined,
+  variant: 'text',
+  text: 'Label',
+  disabled: false,
+  helpertext: '',
+  modelValue: '',
+  size: 'md',
+});
+
+const accordianRef: Ref = ref<boolean>(false);
+slots.sentContent ? (accordianRef.value = true) : (accordianRef.value = false);
+const size: Ref = ref<string>(props.size);
+const variantMiddleware: Ref = ref<string>(props.variant);
+const showPassword: Ref = ref<boolean>(false);
+const type: Ref = ref<string>(props.variant);
+const accordianSwitch: Ref = ref<boolean>(false);
+
+defaultVariantMixin(inputSize).verifyVariant(size.value)
+  ? ''
+  : (size.value = 'md');
+
+defaultVariantMixin(inputVariant).verifyVariant(props.variant)
+  ? ''
+  : (type.value = 'text');
+
+const inputClass: Ref = ref('input--' + type.value);
+
+// Watch for Password-show
+watch(
+  () => showPassword.value,
+  (newVal) => {
+    newVal
+      ? (variantMiddleware.value = 'text')
+      : (variantMiddleware.value = 'password');
+  }
+);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function emitInput(e: any) {
+  const targetVal: string = e.value;
+  emit('update:modelValue', targetVal);
+}
+
+function changePasswordVisibility() {
+  showPassword.value = !showPassword.value;
+}
+
+function toggleAccordian() {
+  accordianSwitch.value = !accordianSwitch.value;
+}
+
+function start(el: HTMLElement) {
+  el.style.height = el.scrollHeight + 'px';
+}
+function end(el: HTMLElement) {
+  el.style.height = '';
+}
+
+// dynamic component import
+const AsyncIcon = computed(() => {
+  {
+    const Icon = defineAsyncComponent(
+      () => import('@/components/icons/icons.vue')
+    );
+    return Icon;
+  }
+});
+
+//USE sentContent class for styling in parent
+</script>
+
+<template>
+  <div class="inputBody">
+    <div :class="[size, 'topWrapper', props.disabled ? 'disabled' : '']">
+      <h1>
+        {{ props.text }}
+        <div v-if="accordianRef" class="iconWrapper">
+          <AsyncIcon
+            v-if="accordianRef"
+            class="dropdown h2"
+            :class="[accordianSwitch ? 'toggledAccordian' : '']"
+            icon="caret-down"
+            @click="toggleAccordian"
+          />
+        </div>
+      </h1>
```

Comment:
I left a comment in the number input that correlates to this one aswell. Use label and then name in the input to connect them!

Reply:
Good with label, same as with-number input about the font-sizes here!